cmd/service/metro/metro.go

```diff
@@ -80,24 +80,22 @@ func Run(ctx context.Context) {
 		}
 	}()
 
-	errChan := component.Start()
+	err := component.Start()
```

Comment:
The component here can be encapsulated into a struct along with ctx. This would avoid passing the same ctx twice i.e in Init() and Run()

Reply:
since errCh was removed from interface, because of that change this code was breaking, this was temp fix to let the build pass and fix rest things, updated

Reply:
using same ctx in multiple func should be fine, will think if we need a struct here